CMakeLists.txt

```diff
@@ -170,38 +161,24 @@ else()
 	add_definitions(-D_ARCH_32=1)
 endif()
 
-include(CheckCCompilerFlag)
 if(ENABLE_GENERIC)
 	message(STATUS "Warning! Building generic build!")
 	set(_M_GENERIC 1)
 	add_definitions(-D_M_GENERIC=1)
-elseif(_ARCH_64 AND (
-	${CMAKE_SYSTEM_PROCESSOR} MATCHES "^x86" OR
-	${CMAKE_SYSTEM_PROCESSOR} MATCHES "i.86" OR
-	${CMAKE_SYSTEM_PROCESSOR} MATCHES "amd64" OR
-	APPLE
-))
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64|AMD64")
 	set(_M_X86 1)
 	set(_M_X86_64 1)
-	add_definitions(-D_M_X86=1 -D_M_X86_64=1 -msse2)
-	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fno-pie")
-	if(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
-		CHECK_C_COMPILER_FLAG("-no-pie" NO_PIE_UPSTREAM)
-		if(NO_PIE_UPSTREAM)
-			set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -no-pie")
-		endif()
-		CHECK_C_COMPILER_FLAG("-nopie" NO_PIE_PATCHED)
-		if(NO_PIE_PATCHED)
-			set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -nopie")
-		endif()
-	endif()
-elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "aarch64")
+	add_definitions(-D_M_X86=1)
+	add_definitions(-D_M_X86_64=1)
+	check_and_add_flag(HAVE_SSE2 -msse2)
```

Comment:
And it's not a problem, but SSE2 is a requirement for amd64 cpu, so we might as well enable the instruction set.